packages/user/src/models/invite.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  PrimaryKey,
+  Property,
+  Index,
+  Filter,
+  OptionalProps,
+} from "@mikro-orm/core"
+
+import { DALUtils, generateEntityId } from "@medusajs/utils"
+import { DAL } from "@medusajs/types"
+
+type OptionalFields =
+  | "metadata"
+  | "accepted"
+  | DAL.SoftDeletableEntityDateColumns
+@Entity({ tableName: "invite" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export default class Invite {
+  [OptionalProps]: OptionalFields
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Index({
+    name: "invite_user_identifier_index",
+    expression: `create unique index "IDX_invite_email" on "invite" ("email") where deleted_at is null;`,
+  })
+  @Property({ columnType: "text" })
+  email: string
```

Comment:
todo: email is probably a column that can be used for querying, I suggest to have an index on it

Reply:
maybe same for token let me know